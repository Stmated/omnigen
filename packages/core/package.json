{
  "name": "@omnigen/core",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "scripts": {
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
    "build": "vite build --config vite.config.js",
    "build:profile": "vite build --config vite.config.profile.ts",
    "dev": "tsc -w",
    "lint": "TIMING=1 eslint src/**/*.ts* --fix",
    "test": "vitest run",
    "test:watch": "vitest"
  },
  "dependencies": {
    "@etclabscore/string-json-template-language": "^1.0.0",
    "@omnigen/api": "workspace:*",
    "@omnigen/core-log": "workspace:*",
    "@omnigen/core-plugin": "workspace:*",
    "@open-rpc/schema-utils-js": "^1.16.2",
    "change-case": "^5.4.4",
    "deepmerge": "^4.3.1",
    "js-yaml": "^4.1.0",
    "json-pointer": "^0.6.2",
    "node-fetch": "^3.3.2",
    "object-hash": "^3.0.0",
    "string-similarity": "^4.0.4",
    "sync-fetch": "^0.5.2",
    "ts-json-schema-generator": "^1.5.1",
    "yaml": "^2.8.1",
    "zod": "^3.25.76"
  },
  "devDependencies": {
    "@omnigen/config-typescript": "workspace:*",
    "@omnigen/eslint-config-custom": "workspace:*",
    "@swc/core": "^1.13.3",
    "@types/json-pointer": "^1.0.34",
    "@types/json-schema": "^7.0.15",
    "@types/node": "^18.19.122",
    "@types/object-hash": "^2.2.1",
    "@types/string-similarity": "^4.0.2",
    "@types/sync-fetch": "^0.4.3",
    "colors": "^1.4.0",
    "nock": "^13.5.6",
    "terser": "^5.43.1",
    "typescript": "5.9.2"
  }
}
