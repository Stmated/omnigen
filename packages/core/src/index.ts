import {PluginAutoRegistry} from '@omnigen/core-plugin';
import {CommonTransformPlugin, CommonTransform2Plugin, fileWriter, CorePlugin} from './CoreUtilPluginInit.ts';

export * from './ast';
export * from './equality';
export * from './parse';
export * from './util';
export * from './visit';
export * from './write';
export * from './CoreUtilPluginInit.ts';

export default PluginAutoRegistry.register([CorePlugin, CommonTransformPlugin, CommonTransform2Plugin, fileWriter]);
