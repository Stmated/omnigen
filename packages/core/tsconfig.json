{
  "$schema": "https://json.schemastore.org/tsconfig",
  "extends": "@omnigen/config-typescript/build.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "baseUrl": "./src"
  }
}
