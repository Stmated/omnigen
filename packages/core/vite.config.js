import {resolve} from 'path';
import {defineConfig} from 'vite';

export default defineConfig({
  build: {
    outDir: 'dist/lib',
    lib: {
      // Could also be a dictionary or array of multiple entry points
      entry: resolve(__dirname, 'src/index.ts'),
      name: '@omnigen/core',
      // the proper extensions will be added
      // fileName: 'core',
    },
    rollupOptions: {
      // make sure to externalize deps that shouldn't be bundled into your library
      external: ['node-fetch', 'promisify', 'node:fs', 'node:url', 'find-up', 'fs/promises', 'fs'],
      output: {
        // Provide global variables to use in the UMD build for externalized deps
        // globals: {
        //   vue: 'Vue',
        // },
      },
      treeshake: 'safest',
    },
  },
  optimizeDeps: {
    exclude: ['sync-fetch'],
  },
});
