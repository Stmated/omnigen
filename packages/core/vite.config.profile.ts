import {resolve} from 'path';
import {type UserConfig} from 'vite';

export default {
  build: {
    outDir: 'dist/profile',
    minify: false,
    terserOptions: {
      compress: false,
      mangle: false,
    },
    lib: {
      // Could also be a dictionary or array of multiple entry points
      entry: resolve(__dirname, 'src/index.profiling.ts'),
      name: '@omnigen/core',
      // the proper extensions will be added
      // fileName: 'core',
    },
    rollupOptions: {
      external: ['node-fetch', 'promisify', 'node:fs', 'node:url', 'find-up'],
      output: {
        // Provide global variables to use in the UMD build for externalized deps
        // globals: {
        //   vue: 'Vue',
        // },
      },
    },
  },
} satisfies UserConfig;
