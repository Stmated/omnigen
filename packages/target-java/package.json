{
  "name": "@omnigen/target-java",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "exports": {
    "ast": "./src/ast/index.ts",
    "default": "./src/index.ts"
  },
  "scripts": {
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
    "dev": "tsc -w",
    "lint": "TIMING=1 eslint src/**/*.ts* --fix",
    "test": "vitest run",
    "test:watch": "vitest"
  },
  "dependencies": {
    "@omnigen/api": "workspace:*",
    "@omnigen/core": "workspace:*",
    "@omnigen/core-log": "workspace:*",
    "@omnigen/core-plugin": "workspace:*",
    "@omnigen/plugin": "workspace:*",
    "@omnigen/target-code": "workspace:*",
    "graphology": "^0.25.4",
    "graphology-dag": "0.4.1",
    "zod": "^3.25.76"
  },
  "devDependencies": {
    "@omnigen/config-typescript": "workspace:*",
    "@omnigen/eslint-config-custom": "workspace:*",
    "@omnigen/utils-test": "workspace:*",
    "java-parser": "^3.0.1",
    "typescript": "5.9.2"
  }
}
