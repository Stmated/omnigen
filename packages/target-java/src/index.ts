import {OMNI_GENERIC_FEATURES, TargetFeatures} from '@omnigen/core';

export * from './visit/index.js';
export * from './transform/index.js';
export * from './interpret/index.js';
export * from './options/index.js';
export * from './util/index.js';
export * from './render/index.js';
export * from './parse/index.js';
export * as Java from './ast/index.js';

export const JAVA_FEATURES: Readonly<TargetFeatures> = {
  ...OMNI_GENERIC_FEATURES,
  literalTypes: false,
  primitiveGenerics: false,
};
