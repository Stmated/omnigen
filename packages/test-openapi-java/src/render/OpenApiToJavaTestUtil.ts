import {RenderedCompilationUnit, TargetOptions} from '@omnigen/api';
import {PluginManager} from '@omnigen/plugin';
import {JavaOptions, JavaPlugins, SerializationLibrary} from '@omnigen/target-java';
import {LoggerFactory} from '@omnigen/core-log';
import {OpenApiPlugins} from '@omnigen/parser-openapi';
import {FileContext} from '@omnigen/core-plugin';
import {DEFAULT_SPECIFIC_TEST_TARGET_OPTIONS} from '@omnigen/utils-test';

const logger = LoggerFactory.create(import.meta.url);

export const DEFAULT_SPECIFIC_TEST_JAVA_OPTIONS: Partial<JavaOptions> = {
  serializationLibrary: SerializationLibrary.POJO,
  beanValidation: false,
};

export class OpenApiToJavaTestUtil {

  public static async render(inPath: string, options?: Partial<JavaOptions & TargetOptions>): Promise<RenderedCompilationUnit[]> {

    const pm = new PluginManager({includeAuto: true});

    logger.info(`Running plugin manager with ${OpenApiPlugins.OpenApiPlugin.name} and ${JavaPlugins.JavaPluginInit.name}`);

    const ctx: FileContext = {
      file: inPath,
      defaults: {},
      arguments: {
        ...DEFAULT_SPECIFIC_TEST_JAVA_OPTIONS,
        ...DEFAULT_SPECIFIC_TEST_TARGET_OPTIONS,
        ...options,
      },
    };

    const result = await pm.execute({
      ctx: ctx,
      stopAt: JavaPlugins.JavaRendererCtxOut,
    });

    return result.result.ctx.compilationUnits;
  }
}
