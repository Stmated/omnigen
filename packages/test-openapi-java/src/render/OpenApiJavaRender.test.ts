import {describe, test, vi} from 'vitest';
import {OpenApiToJavaTestUtil} from './OpenApiToJavaTestUtil.ts';
import {SerializationLibrary} from '@omnigen/target-java';
import {Util} from '@omnigen/core';
import {TestUtils} from '@omnigen/utils-test';

describe('openapi-java-render', () => {

  test('petstore', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await OpenApiToJavaTestUtil.render(Util.getPathFromRoot('./packages/parser-openapi/examples/petstore.yaml'), {
      serializationLibrary: SerializationLibrary.POJO,
      // debug: true,
    });
    const fileContents = Map.groupBy(rendered, it => it.fileName);

    ctx.expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, cu] of fileContents) {
      ctx.expect(cu).toHaveLength(1);
      await ctx.expect(cu[0].content).toMatchFileSnapshot(TestUtils.getSnapshotFileName(ctx, fileName));
    }
  });
});
