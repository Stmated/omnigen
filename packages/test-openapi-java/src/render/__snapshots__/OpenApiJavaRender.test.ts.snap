// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`openapi-java-render > petstore 1`] = `
[
  "Cat.java",
  "Dog.java",
  "Pet.java",
]
`;
